local-tests/setup/tinny-environment.ts

```diff
@@ -0,0 +1,366 @@
+      '0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6',
+      '0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a',
+      '0x8b3a350cf5c34c9194ca85829a2df0ec3153be0318b5e2d3348e872092edffba',
+      '0x92db14e403b83dfe3df233f83dfa3a0d7096f21ca9b0d6d6b8d88b2b4ec1564e',
+      '0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356',
+      '0xdbda1821b80551c9d65939329250298aa3472ba22feea921c0cf5d620ea67b97',
+      '0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6',
+    ],
+    KEY_IN_USE: new Array(),
+  };
+
+  public litNodeClient: LitNodeClient;
+  public contractsClient: LitContracts;
+  public rpc: string;
+  public superCapacityDelegationAuthSig: AuthSig;
+
+  constructor(network?: LIT_TESTNET) {
+    // -- setup networkj
+    this.network = network || this.processEnvs.NETWORK;
+
+    if (Object.values(LIT_TESTNET).indexOf(this.network) === -1) {
+      throw new Error(
+        `Invalid network environment. Please use one of ${Object.values(
+          LIT_TESTNET
+        )}`
+      );
+    }
+
+    // -- create an empty array to keep track of all keys
+    this.processEnvs.KEY_IN_USE = new Array(
+      this.processEnvs.PRIVATE_KEYS.length
+    ).fill(false);
+
+    // -- setup rpc
+    if (this.network === LIT_TESTNET.LOCALCHAIN) {
+      this.rpc = this.processEnvs.LIT_RPC_URL;
+    } else {
+      this.rpc = this.processEnvs.LIT_OFFICIAL_RPC;
+    }
+  }
+
+  world: Map<string, TinnyPerson> = new Map();
+
+  /**
+   * Retrieves an available private key from a list, marking it as in use and scheduling
+   * its automatic release. If no unused keys are available, it waits for a set interval
+   * before rechecking.
+   *
+   * This function loops until it finds an unused key, marks it, and returns the key with
+   * its index. If all keys are in use, it logs a wait message and pauses before retrying.
+   *
+   * Outputs:
+   * - privateKey: The selected private key.
+   * - index: The index of the selected key.
+   *
+   * Environment variables required:
+   * - KEY_IN_USE: Boolean array indicating key usage.
+   * - PRIVATE_KEYS: Array of key strings.
+   * - TIME_TO_RELEASE_KEY: Milliseconds until a key is automatically released.
+   * - WAIT_FOR_KEY_INTERVAL: Wait time in milliseconds if no keys are free.
+   */
+  async getAvailablePrivateKey(): Promise<{
+    privateKey: string;
+    index: number;
+  }> {
+    while (true) {
+      const index = this.processEnvs.KEY_IN_USE.findIndex((used) => !used); // Find the first unused key
+
+      if (index !== -1) {
+        // If an available key is found
+        this.processEnvs.KEY_IN_USE[index] = true; // Mark the key as in use
+        console.log('[𐬺🧪 Tinny Environment𐬺] 🔑 Selected key at index', index); // Log a message indicating that we have selected a key
+
+        // Set a timer to automatically release the key after 10 seconds
+        setTimeout(() => {
```

Comment:
We do we release the key after a set amount of time?  Depending on conditions, it might take longer than that to complete our operations, leading to a key that is still being used being treated as 'available', and thus breaking both the existing task and the new task that needed a key due to nonce errors; if code must call `getAvailablePrivateKey()`, why doesn't it call `releasePrivateKey()` explicitly when it is done?